### DISCUSSION TITLE
[Data Loaders] Reconcider addition of `dataLoader`/`dataPageLoader` SFC macros

### DISCUSSION THREAD

**Discussion by rijenkii** [posva/unplugin-vue-router#553] [Questions and Help] [answered]
https://github.com/posva/unplugin-vue-router/blob/ef056ab2b8290bdee7554966ba82311d8423e99c/docs/data-loaders/rfc.md?plain=1#L849-L852

The requirement of using non-setup `<script>` to define pages with loaders is the *only* thing that turns me away from using data loaders in my projects.

Vanilla Vue already provides macros that accept local variables from `<script setup>`, hoists them and actively screams at the user if they did something egregiously wrong, so data loader macros could do the same:
```vue
<script setup lang="ts">
import { ref } from "vue";

const inheritProps = ref(true);
defineOptions({ inheritProps });
</script>
```
```
[vite:vue] [@vue/compiler-sfc] `defineOptions()` in <script setup> cannot reference locally declared variables because it will be hoisted outside of the setup() function. If your component options require initialization in the module scope, use a separate normal <script> to export the options instead.

/project/workspace/src/App.vue
3  |  
4  |  const inheritProps = ref(true);
5  |  defineOptions({ inheritProps });
   |                  ^^^^^^^^^^^^
6  |  </script>
7  |  
```

**Comment by posva** [comment, marked as answer]
I know 😅
Currently there is an experimental setting that automatically adds the export. I can’t remember the name, but it should be easy to find with auto completion on the experimental option

**Reply by rijenkii** [reply]
Thank you for the answer. Linked usage examples in the RFC comments to calm people down: https://github.com/vuejs/rfcs/discussions/460#discussioncomment-11564225
